newton/examples/__init__.py

```
@@ -45,8 +173,34 @@ def run(example):
 
+    if args is not None and args.test and hasattr(example, "test"):
+        example.test()
+
     example.viewer.close()
 
+    if args is not None and args.test:
+        # generic tests for finiteness of Newton objects
+        if hasattr(example, "state_0"):
+            nan_members = _find_nan(example.state_0)
+            if nan_members:
+                raise ValueError(f"NaN members found in state_0: {nan_members}")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)